pkg/controllers/clusterresourceplacement/resource_selector.go

```diff
@@ -330,7 +359,8 @@ func (r *Reconciler) fetchNamespaceResources(selector fleetv1beta1.ClusterResour
 	}
 	namespaces, err := r.InformerManager.Lister(utils.NamespaceGVR).List(labelSelector)
 	if err != nil {
-		return nil, controller.NewAPIServerError(true, fmt.Errorf("cannot list all the namespaces given the label selector: %w", err))
+		klog.ErrorS(err, "Cannot list all the namespaces by the label selector", "labelSelector", labelSelector, "placement", placeName)
+		return nil, controller.NewAPIServerError(true, err)
```

Comment:
NewAPIServerError already does a klog.error, why print it again?

Reply:
You're right, validated the chain behavior of the as/is func. https://go.dev/play/p/nyuQ1UFtRAG

errors.Is(err, target): This function checks if err or any error in its chain matches target. It recursively unwraps the error chain until a match is found or no more errors can be unwrapped.
errors.As(err, target): This function checks if err or any error in its chain matches the type of target. If a match is found, it assigns the error from the chain to target.

Before i hit a UT failure, i guess it should not be related to the wrapper here.